From 46cc70400b88279b050de4dec9e639c8d14f137e Mon Sep 17 00:00:00 2001
From: Andrei Botila <andrei.botila@nxp.com>
Date: Tue, 8 Feb 2022 13:49:03 +0200
Subject: [PATCH 3/9] s32: remove MC_ME function related to secondary cores

Removed mc_me_get_cluster_ptrn(), get_rgm_a53_bit() and
mc_me_core2prtn_core_id().

Issue: ALB-8344
Upstream-Status: Pending 

Signed-off-by: Andrei Botila <andrei.botila@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/s32-gen1/s32g3_soc.c   | 74 -------------------
 arch/arm/cpu/armv8/s32/s32-gen1/soc.c         | 24 ------
 .../asm/arch-s32/s32-gen1/mc_me_regs.h        |  7 +-
 .../asm/arch-s32/s32-gen1/mc_rgm_regs.h       |  7 +-
 4 files changed, 4 insertions(+), 108 deletions(-)
 delete mode 100644 arch/arm/cpu/armv8/s32/s32-gen1/s32g3_soc.c

diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/s32g3_soc.c b/arch/arm/cpu/armv8/s32/s32-gen1/s32g3_soc.c
deleted file mode 100644
index f2a727c8bc..0000000000
--- a/arch/arm/cpu/armv8/s32/s32-gen1/s32g3_soc.c
+++ /dev/null
@@ -1,74 +0,0 @@
-// SPDX-License-Identifier:     GPL-2.0+
-/*
- * Copyright 2021 NXP
- */
-
-#include <asm/arch/s32-gen1/mc_me_regs.h>
-
-u8 mc_me_core2prtn_core_id(u8 part, u8 id)
-{
-	/**
-	 * A map where the key is core id obtained from MPIDR and the
-	 * value represents the ID of the core in MC_ME.PRTN1_CORE*
-	 */
-	static const u8 mc_me_a53_core_id[] = {
-		/* Cluster 0, core 0 */
-		[0] = 0,
-		/* Cluster 0, core 1 */
-		[1] = 1,
-		/* Cluster 0, core 2 */
-		[2] = 4,
-		/* Cluster 0, core 3 */
-		[3] = 5,
-		/* Cluster 1, core 0 */
-		[4] = 2,
-		/* Cluster 1, core 1 */
-		[5] = 3,
-		/* Cluster 1, core 2 */
-		[6] = 6,
-		/* Cluster 1, core 3 */
-		[7] = 7,
-	};
-
-	static const u8 mc_me_m7_core_id[] = {
-		/* Core 0 */
-		[0] = 0,
-		[1] = 1,
-		[2] = 2,
-		/* Core 3 */
-		[3] = 4,
-	};
-
-	if (part == MC_ME_CORES_PRTN)
-		return mc_me_a53_core_id[id];
-
-	return mc_me_m7_core_id[id];
-}
-
-u8 get_rgm_a53_bit(u8 core)
-{
-	static u8 periph_rgm_cores[] = {
-		/** Cluster 0, core 0*/
-		[0] = 65,
-		[1] = 66,
-		[2] = 69,
-		[3] = 70,
-		/** Cluster 1, core 0*/
-		[4] = 67,
-		[5] = 68,
-		[6] = 71,
-		[7] = 72,
-	};
-
-	return periph_rgm_cores[core] % 64;
-}
-
-u32 mc_me_get_cluster_ptrn(u32 core)
-{
-	/**
-	 * For G32G3 we have the following mapping:
-	 *     MC_ME_PRTN1_CORE0_* -> CA53 cluster0 core0/1/2/3
-	 *     MC_ME_PRTN1_CORE2_* -> CA53 cluster1 core0/1/2/3
-	 */
-	return mc_me_core2prtn_core_id(MC_ME_CORES_PRTN, core) & 2;
-}
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
index 0ab8c072e7..5c843d8a08 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
@@ -190,27 +190,3 @@ __weak u32 get_sram_size(void)
 {
 	return S32_SRAM_SIZE;
 }
-
-__weak u8 mc_me_core2prtn_core_id(u8 part, u8 id)
-{
-	return id;
-}
-
-__weak u8 get_rgm_a53_bit(u8 core)
-{
-	/**
-	 * Bit corresponding to CA53_n in the cores'
-	 * RGM reset partition (n=0..3)
-	 */
-	return core + 1;
-}
-
-__weak u32 mc_me_get_cluster_ptrn(u32 core)
-{
-	/**
-	 * For S32G2/S32R45 we have the following mapping:
-	 *     MC_ME_PRTN1_CORE0_* -> CA53 cluster0 core0/1
-	 *     MC_ME_PRTN1_CORE2_* -> CA53 cluster1 core0/1
-	 */
-	return (core % 4) & ~1;
-}
diff --git a/arch/arm/include/asm/arch-s32/s32-gen1/mc_me_regs.h b/arch/arm/include/asm/arch-s32/s32-gen1/mc_me_regs.h
index f0a6a24591..1fa14561dc 100644
--- a/arch/arm/include/asm/arch-s32/s32-gen1/mc_me_regs.h
+++ b/arch/arm/include/asm/arch-s32/s32-gen1/mc_me_regs.h
@@ -41,8 +41,8 @@
 #define MC_ME_PRTN_PART(PART, PRTN)	(MC_ME_BASE_ADDR + 0x140UL + \
 					 (PART) * 0x200UL + \
 					 (PRTN) * 0x20UL)
-#define MC_ME_PRTN_N_CORE_M(n, m)	\
-	MC_ME_PRTN_PART(n, mc_me_core2prtn_core_id((n), (m)))
+#define MC_ME_PRTN_N_CORE_M(n, m)      \
+	MC_ME_PRTN_PART(n, m)
 
 #define MC_ME_PRTN_N_PCONF_OFF	0x0
 #define MC_ME_PRTN_N_PUPD_OFF	0x4
@@ -78,9 +78,6 @@
 
 #define RDC_RD_STAT_XBAR_DISABLE_MASK	BIT(4)
 
-u8 mc_me_core2prtn_core_id(u8 part, u8 id);
-u32 mc_me_get_cluster_ptrn(u32 core);
-
 #endif
 
 #endif /*__ARCH_ARM_MACH_S32GEN1_MCME_REGS_H__ */
diff --git a/arch/arm/include/asm/arch-s32/s32-gen1/mc_rgm_regs.h b/arch/arm/include/asm/arch-s32/s32-gen1/mc_rgm_regs.h
index 8093d77e97..83b6244a7d 100644
--- a/arch/arm/include/asm/arch-s32/s32-gen1/mc_rgm_regs.h
+++ b/arch/arm/include/asm/arch-s32/s32-gen1/mc_rgm_regs.h
@@ -1,8 +1,7 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
 /*
  * (C) Copyright 2015-2016 Freescale Semiconductor, Inc.
- * Copyright 2016-2020 NXP
- *
- * SPDX-License-Identifier:	GPL-2.0+
+ * Copyright 2016-2020,2022 NXP
  */
 
 #ifndef __ARCH_ARM_MACH_S32GEN1_MCRGM_REGS_H__
@@ -45,7 +44,5 @@
 #define PRST_PCIE_1_FUNC		17
 #define PRST_PFE			128
 
-u8 get_rgm_a53_bit(u8 core);
-
 #endif /* __ARCH_ARM_MACH_S32GEN1_MCRGM_REGS_H__ */
 
-- 
2.17.1

